q2_cutadapt/tests/test_trim.py

```diff
@@ -0,0 +1,194 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import unittest
+
+from q2_cutadapt._trim import _build_trim_command
+from q2_types.per_sample_sequences import (
+    CasavaOneEightSingleLanePerSampleDirFmt,
+    SingleLanePerSampleSingleEndFastqDirFmt,
+    SingleLanePerSamplePairedEndFastqDirFmt,
+    FastqGzFormat,
+)
+from qiime2 import Artifact
+from qiime2.util import redirected_stdio
+from qiime2.plugin.testing import TestPluginBase
+
+
+class TestTrimSingle(TestPluginBase):
+    package = 'q2_cutadapt.tests'
+
+    # This test is really just to make sure that the command runs - the
+    # detailed tests in the Util Tests below ensure the commands are crafted
+    # appropriately.
+    def test_typical(self):
+        demuxed = Artifact.import_data('SampleData[SequencesWithQuality]',
+                                       self.get_data_path('single-end'))
+        with redirected_stdio(stdout=os.devnull):
+            self.plugin.methods['trim_single'](demuxed)
+        # TODO: test results
+
+
+class TestTrimPaired(TestPluginBase):
+    package = 'q2_cutadapt.tests'
+
+    # This test is really just to make sure that the command runs - the
+    # detailed tests in the Util Tests below ensure the commands are crafted
+    # appropriately.
+    def test_typical(self):
+        demuxed = Artifact.import_data(
+            'SampleData[PairedEndSequencesWithQuality]',
+            self.get_data_path('paired-end'))
+        with redirected_stdio(stdout=os.devnull):
+            self.plugin.methods['trim_paired'](demuxed)
+        # TODO: test results
+
+
+class TestTrimUtilsSingle(TestPluginBase):
+    package = 'q2_cutadapt.tests'
+
+    def setUp(self):
+        super().setUp()
+
+        self.demux_seqs = SingleLanePerSampleSingleEndFastqDirFmt(
+            self.get_data_path('single-end'), mode='r')
+        self.trimmed_seqs = CasavaOneEightSingleLanePerSampleDirFmt()
+
+    def test_build_trim_command_typical(self):
```

Comment:
Does this cover overriding all of the default parameters? I like to do that, to make sure I haven't mixed up and parameter/value pairs, and that all parameters are being passed.